k8s-scheduler/src/main/java/com/vmware/dcm/Scheduler.java

```diff
@@ -457,22 +466,35 @@ private void requeue(final Result<? extends Record> unassignedPods) {
         return podsToAssignUpdated;
     }
 
-    Result<? extends Record> scopedFunction(final Set<String> augViews) {
+    Result<? extends Record> scopedFunction(final AutoScopeViews autoScopeViews) {
         final Timer.Context solveTimer = solveTimes.time();
+        final Set<String> augViews = autoScopeViews.augmentedViews().keySet();
         final Result<? extends Record> podsToAssignUpdated = initialPlacement.solve(
                 "PODS_TO_ASSIGN", (t) -> {
-                    final DSLContext conn = dbConnectionPool.getConnectionToDb();
                     final String augView = (t.getName() + SCOPE_VIEW_NAME_SUFFIX).toUpperCase();
                     final Table<?> toFetch;
                     if (augViews.contains(augView)) {
-                        LOG.info(String.format("Scoping Optimization: Replace %s with %s", t.getName(), augView));
                         toFetch = table(augView);
                     } else {
                         toFetch = t;
                     }
                     if (dbConnectionPool instanceof DDlogDBConnectionPool) {
-                        return ((DDlogDBConnectionPool) dbConnectionPool).getProvider()
-                                .fetchTable(toFetch.getName());
+                        final DDlogJooqProvider provider = ((DDlogDBConnectionPool) dbConnectionPool).getProvider();
+                        final Result<Record> augResult = provider.fetchTable(toFetch.getName());
```

Comment:
I might be missing something. Wouldn't `augResult` in line 483 already have all the rows in it (that is, the query without the topK filter)? And if so, wouldn't the condition in line 488 always be true?

Reply:
Before this PR, in AutoScope, we create spare_capacity_per_node_augment = (domain restricting queries) UNION (top K queries). Right now in AutoScope, the augment view only contains the domain restricting conditions and we are performing the union in the for loop on line 487. Therefore line 488 is not always true.

Reply:
Ah, got it. Thanks.

In my head I was thinking of the solution where we were applying topK to multiple domain restricting views, and union-ing the results. This is not that case, but is instead, as you say, domain restricting views UNION topK.